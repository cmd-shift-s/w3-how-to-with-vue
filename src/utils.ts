import MarkdownIt from 'markdown-it'
import hljs from 'highlight.js'
import 'highlight.js/styles/vs.css'

export const md = new MarkdownIt({
  highlight (code, lang) {
    return hljs.highlight(lang, code).value
  }
})
