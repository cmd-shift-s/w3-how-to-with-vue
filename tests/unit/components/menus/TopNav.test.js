import { mount, renderer } from '@/tests/unit/utils'
import TopNav from '@/components/menus/TopNav.vue'

describe('TopNav.vue', () => {
  it('render default', () => {
    const wrap = mount(TopNav)

    expect(wrap.classes()).toContain('top-nav')

    renderer.renderToString(wrap.vm, (err, str) => {
      if (err) console.error(err)
      expect(str).toMatchSnapshot()
    })
  })

  it('slots#default', () => {
    const wrap = mount(TopNav, {
      slots: {
        default: '<a>Test</a>'
      }
    })

    expect(wrap.classes()).toContain('top-nav')
    expect(wrap.find('a').text()).toEqual('Test')

    renderer.renderToString(wrap.vm, (err, str) => {
      if (err) console.error(err)
      expect(str).toMatchSnapshot()
    })
  })

  it('props#is-responsive', () => {
    const wrap = mount(TopNav, {
      propsData: {
        isResponsive: true
      }
    })

    expect(wrap.classes()).toContains(['top-nav', 'is-responsive'])

    renderer.renderToString(wrap.vm, (err, str) => {
      if (err) console.error(err)
      expect(str).toMatchSnapshot()
    })

    wrap.vm.$el.querySelector('.toggle-menu').click()
    wrap.update()

    expect(wrap.classes()).toContain('is-active')

    renderer.renderToString(wrap.vm, (err, str) => {
      if (err) console.error(err)
      expect(str).toMatchSnapshot()
    })
  })

  it('props#is-bottom', () => {
    const wrap = mount(TopNav, {
      propsData: {
        isBottom: true
      }
    })

    expect(wrap.classes()).toContains(['top-nav', 'is-bottom'])

    renderer.renderToString(wrap.vm, (err, str) => {
      if (err) console.error(err)
      expect(str).toMatchSnapshot()
    })
  })
})
