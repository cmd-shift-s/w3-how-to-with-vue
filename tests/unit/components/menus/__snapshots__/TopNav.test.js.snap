// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopNav.vue props#is-bottom 1`] = `
<nav class="top-nav is-bottom">
    <!---->
</nav>
`;

exports[`TopNav.vue props#is-responsive 1`] = `<nav class="top-nav is-responsive"> <a class="toggle-menu">☰</a></nav>`;

exports[`TopNav.vue props#is-responsive 2`] = `<nav class="top-nav is-responsive is-active"> <a class="toggle-menu">☰</a></nav>`;

exports[`TopNav.vue render default 1`] = `
<nav class="top-nav">
    <!---->
</nav>
`;

exports[`TopNav.vue slots#default 1`] = `
<nav class="top-nav"><a>Test</a>
    <!---->
</nav>
`;
